### ISSUE TITLE
Add time out functionality to `select_on_a_map`

### ISSUE THREAD

**Issue by robbibt** [opendatacube/odc-tools#31]
Adding a time out functionality to `select_on_a_map` would allow us to implement testing on functions and notebooks that use interactive map selection. For example, if no user feedback was observed after X seconds, fall back to a default bounding box extent.

**Comment by robbibt** [comment]
@Kirill888

**Comment by Kirill888** [comment]
@robbibt what would be a good way to specify default value to return?
The output of `select_on_a_map` is a `Geometry` class which is a bit too painful to construct.

Options:

1. Bounding box as a tuple of 4 values in lonlat notation compatible with `datacube.utils.geometry.BoundingBox` class

```python
extents = select_on_a_map(default=(left, bottom, right, top))
```

2. Dictionary of `x|y` or `lon|lat` ranges

```python
extents = select_on_a_map(default={'x': (left, right), 
                                   'y': (bottom, top)})
```

To make is easier for testing I suggest following rules for timeout behaviour:

1. If both `default=` is set and `timeout=` are supplied
    - Return value supplied in `default=` converted to `Geometry` after timeout is reached
    - OR return user selection if happens quickly enough 
2. If `timeout=` is set but not `default=` 
    - Raise Exception when timeout is reached
3. If `default=` is set but not `timeout=`
   - Check env variable `ODC_MAP_SELECT_TIMEOUT=` for the value of `timeout=`: block if unset, otherwise behave as in (1)

**Comment by robbibt** [comment]
Hmm, that's a great question. Ideally we wouldn't want to create too much extra handling code further downstream for when the default is returned... does `dc.load` allow data to be loaded using the `BoundingBox` class or would it have to be converted into a `Geometry` object before it can be used?

Would it be possible to have a simple hard-coded `Geometry` object that's defined somewhere within the `select_on_a_map` function so that a `Geometry` object is always returned (and therefore gives valid results in testing without additional handling), but which could be over-written by passing an alternative default? It could be based on the lat-lon that you already have to pass to centre the map...

The timeout behaviour rules make sense to me.

**Comment by robbibt** [comment]
E.g. something like:

```
x, y = 119.51, -20.37

y_min = y - 0.05
y_max = y + 0.05
x_min = x - 0.05
x_max = x + 0.05

from datacube.utils.geometry import Geometry

Geometry({'type': 'Polygon',
 'coordinates': [[(x_min, y_max),
                  (x_min, y_min),
                  (x_max, y_min),
                  (x_max, y_max),
                  (x_min, y_max)]]})
```

**Comment by Kirill888** [comment]
@robbibt I do not think it's feasible to have a global default at the library level. If you want to avoid setting default extents per notebook that uses this function then you are better off augmenting test runner code to replace this function at run-time with something that immediately returns some global constant.

`BoundingBox` can not be used as part of `dc.load` directly as far as I can tell, `Geometry` objects can be used via `geopolygon=` parameter.

**Comment by robbibt** [comment]
Yep, OK. We can always just wrap `select_on_a_map` inside a larger function that converts the default to a `Geometry` then does the rest of the analysis ([similar to the current `run_filmstrips_app` func here](https://github.com/GeoscienceAustralia/dea-notebooks/blob/develop/Scripts/notebookapp_changefilmstrips.py#L36))
